src/main/java/ch/ivyteam/ivy/maven/validate/ValidateMojo.java

```diff
@@ -0,0 +1,67 @@
+package ch.ivyteam.ivy.maven.validate;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.maven.execution.MavenSession;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.project.MavenProject;
+
+@Mojo(name = ValidateMojo.GOAL, requiresProject = true)
+public class ValidateMojo extends AbstractMojo {
+
+  public static final String GOAL = "validate";
+  protected static final String PLUGIN_GROUPID = "com.axonivy.ivy.ci";
+  protected static final String PLUGIN_ARTIFACTID = "project-build-plugin";
+
+  @Parameter(defaultValue = "${session}", readonly = true, required = true)
+  private MavenSession session;
+
+  @Override
+  public void execute() throws MojoExecutionException, MojoFailureException {
+    validateConsistentPluginVersion(session.getAllProjects());
+  }
+
+  protected void validateConsistentPluginVersion(List<MavenProject> projects) throws MojoExecutionException {
+    Map<String, Set<MavenProject>> versionToProjectsMap = new HashMap<>();
+    for (var project : projects) {
+      for (var plugin : project.getBuild().getPlugins()) {
+        if (PLUGIN_GROUPID.equals(plugin.getGroupId()) && PLUGIN_ARTIFACTID.equals(plugin.getArtifactId())) {
+          var version = plugin.getVersion();
+          // Skip checking plug ins that do not have a version
+          if (version == null) {
+            continue;
+          }
+          getLog().debug(PLUGIN_GROUPID + ":" + plugin.getArtifactId() + ":" + version + " configured in " + project);
+          var projectSet = versionToProjectsMap.get(version);
+          if (projectSet == null) {
+            projectSet = new LinkedHashSet<>();
+            versionToProjectsMap.put(version, projectSet);
+          }
+          projectSet.add(project);
+        }
+      }
+    }
+    if (versionToProjectsMap.size() > 1) {
+      var versions = new ArrayList<>(versionToProjectsMap.keySet());
+      Collections.sort(versions);
+      getLog().error("Several versions of project-build-plugins are configured " + versions + ":");
+      for (var version : versions) {
+        getLog().error(version + ":");
+        for (var project : versionToProjectsMap.get(version)) {
+          getLog().error("\t" + project.toString());
+        }
+      }
```

Comment:
I think we should log only one error, but with new-lines in it? Otherwise we get multiple errors for a single problem?

Reply:
Then the output is like:
```
[ERROR] Several versions of project-build-plugins are configured [12.0.1-SNAPSHOT, 12.0.2-SNAPSHOT]:
12.0.1-SNAPSHOT:
	MavenProject: com.axonivy.demo:workflow-demos-test:12.0.2-SNAPSHOT @ /Users/lli/GitWorkspace/market/demo-projects/workflow/workflow-demos-test/pom.xml
12.0.2-SNAPSHOT:
	MavenProject: com.axonivy.demo:workflow-demos:12.0.2-SNAPSHOT @ /Users/lli/GitWorkspace/market/demo-projects/workflow/workflow-demos/pom.xml
```
instead of 
```
[ERROR] Several versions of project-build-plugins are configured [12.0.1-SNAPSHOT, 12.0.2-SNAPSHOT]:
[ERROR] 12.0.1-SNAPSHOT:
[ERROR] 	MavenProject: com.axonivy.demo:workflow-demos-test:12.0.2-SNAPSHOT @ /Users/lli/GitWorkspace/market/demo-projects/workflow/workflow-demos-test/pom.xml
[ERROR] 12.0.2-SNAPSHOT:
[ERROR] 	MavenProject: com.axonivy.demo:workflow-demos:12.0.2-SNAPSHOT @ /Users/lli/GitWorkspace/market/demo-projects/workflo
```
also fine for me

Reply:
the output on the CLI looks a little worse, but IMHO we could risk it in favour of making log-parsers (like the one from jenkins) more intelligent. ONe problem ... one Log line with a severity..